scripts/needs_testrun.py

```diff
@@ -151,15 +189,22 @@ def needs_testrun(suite: str, pr_number: int, sha: t.Optional[str] = None) -> bo
     return bool(matches)
 
 
-def _get_pr_number():
+def _get_pr_number() -> int:
+    # CircleCI
     number = os.environ.get("CIRCLE_PR_NUMBER")
-    if not number:
-        pr_url = os.environ.get("CIRCLE_PULL_REQUEST", "")
-        number = pr_url.split("/")[-1]
-    try:
+    if number is not None:
         return int(number)
-    except ValueError:
-        return 0
+
+    pr_url = os.environ.get("CIRCLE_PULL_REQUEST")
+    if pr_url is not None:
+        return int(pr_url.split("/")[-1])
+
+    # GitLab
+    ref_name = os.environ.get("CI_COMMIT_REF_NAME")
+    if ref_name is not None:
+        return int(github_api("/pulls", {"head": f"datadog:{ref_name}"})[0]["number"])
```

Comment:
This part also fails in the merge queue. The `CI_COMMIT_REF_NAME` is associated with the newly created `mq-working-branch-*` branch, which isn't associated with the original PR. I'm thinking maybe we can do something similar here and do:
```suggestion
    ref_name = os.environ.get("CI_COMMIT_REF_NAME")
    if ref_name is not None:
        try:
            return int(github_api("/pulls", {"head": f"datadog:{ref_name}"})[0]["number"])
        except Exception:
            LOGGER.warning("Could not determine PR number from CI_COMMIT_REF_NAME")
            return 0
```

Reply:
[Tested this in my test branch and it looks like it triggered all tests for the merge queue](https://gitlab.ddbuild.io/DataDog/apm-reliability/dd-trace-py/-/pipelines/48853125), regardless of the fact that I only made changes to dramatiq 🎉

Reply:
I think if we catch the exception like the above we should be fine too 🤞